src/cpp/session/SessionSourceDatabase.cpp

```diff
@@ -716,6 +732,24 @@ void onShutdown(bool)
       LOG_ERROR(error);
 }
 
+void onDocUpdated(boost::shared_ptr<SourceDocument> pDoc)
+{
+   s_idToPath[pDoc->id()] = pDoc->path();
+}
+
+void onDocRemoved(const std::string& id)
+{
+   std::map<std::string, std::string>::iterator it = s_idToPath.find(id);
+   if (it != s_idToPath.end())
+      s_idToPath.erase(it);
+}
+
+void onDocRenamed(const std::string &, 
```

Comment:
Is it possible for a document's ID to change when it's renamed? (why do we get away with ignoring the first parameter?)